### ISSUE TITLE
Some APIs have changed in the iOS app, may need updates

### ISSUE THREAD

**Issue by tmenguy** [cliviu74/wallbox#43]
Hi,

I was probing the Wallbox iOS app, and found some changes, like the may the  set max charging current is sent.
I've updated the doc here : https://github.com/tmenguy/wallboxAPIDoc and added proper handling of power boost and eco smart config in the documentation.
I'll propose a PR to update the lib as soon as the ones  we do have pending are accepted or rejected.

**Comment by cliviu74** [comment]
@tmenguy not sure I understand the changes requested here. The timeout has been closed as "Won't fix" per our discussion in the PR.
Is there anything else you've seen changed? I've done a complete sweep of all the methods and they seem to be working properly.

The ICP changes are pending a change but we'll make it, although I've not a way of testing they're working on my side (I don't have a power diverter installed on mine)

Anything else you'd like to see added?

**Comment by tmenguy** [comment]
Hey, 

Ah? Perhaps we got a misunderstanding on the won’t fix : I proposed a change in the PR for you to see, and in the PR there was also a fix for token refresh, I removed the helper for requests, so if you can have a last look at it would have been great.

In this is issue I was mentioning that I have analyzed the latest iOS app, and stuff are used differently, like the max sharing current for example is set using a new endpoint for charger config. I have updated the walboxAPIDoc accordingly.
The old way seems ok still, just knowing that they may deprecate some of the stuff we are using today.

**Comment by tmenguy** [comment]
And yes to be added : the ecosmart and power boost correct setting (related to the ICP)  and handling and this I can test I have one.

**Comment by tmenguy** [comment]
never mind for the timeout : I open a new one on the timeout and the token refresh.
As for the changed APIs, it is a discussion : the iOS App is not using anymore the v2 endpoints but some new stuff (same for energy price too), so is there a need to switch to those new ones? Or do we keep th old way while it is working?

**Comment by cliviu74** [comment]
I see now, I'll dig around into the iPhone app to see if there's a consistent trend to moving all functions to the new api. If you have any info and you wouldn't mind sharing, that would be great to start collecting the information before making a decision. 
Previously I noticed the wallbox api versioning is not consistent , with v2, v3 and v4 appearing in the uri. It's possible they're just versioning the features?! (i.e. v1 - authhentication, v2 - basic control, v4 - advanced control... and so on).

**Comment by tmenguy** [comment]
Yes as documentation I’m modifying : https://github.com/SKB-CGN/wallbox (through the fork https://github.com/tmenguy/wallboxAPIDoc), see the part about energy cost and max charge current, or the ICP (home sharing)
